cypress/support/step_definitions/getterSetterCalls.md

```diff
@@ -1,132 +1,327 @@
- * `And 'Firebolt' platform responds to '1st party app' 'CLOSEDCAPTION_SETTINGS' set API`
- * `And 'Firebolt' platform responds to '3rd party app' 'CLOSEDCAPTION_SETTINGS' get API`
- * `And 'Firebolt' platform responds to '1st party app' set API`
- * `And 'Firebolt' platform responds to '1st party app' 'CLOSEDCAPTION_SETTINGS' set API with 'INVALID_TYPE_PARAMS'`
-
-## '(.+)' platform (triggers|does not trigger) '(.*?)'(?: '(.*?)')? event(?: with '(.+)')?
-
-### Purpose: Performing a event validation against the source of truth
-
-### Params:
-| Param | Definition |
-| --- | --- |
-| sdk | sdk name |
-| eventExpected | Determines whether the event is expected or not. |
-| appId |app identifier |
-| fireboltCallKey | key name passed to look for firebolt call object in fireboltCallData |
-| errorContent | Doing error content validation when error content object key passed. Ex: 'INVALID_TYPE_PARAMS' |
-
-### Examples:
- * `And 'Firebolt' platform triggers '1st party app' 'CLOSEDCAPTION_SETTINGS' event`
- * `And 'Firebolt' platform triggers '1st party app' event`
- * `And 'Firebolt' platform triggers '3rd party app' 'CLOSEDCAPTION_SETTINGS' event`
- * `And 'Firebolt' platform does not trigger '3rd party app' 'CLOSEDCAPTION_SETTINGS' event`
- * `And 'Firebolt' platform triggers '1st party app' event`
- * `And 'Firebolt' platform triggers '1st party app' 'CLOSEDCAPTION_SETTINGS' event with 'INVALID_TYPE_PARAMS'`
+## Background
+
+FCS supports the creation of Firebolt call objects, which include methods, parameters, and content. These objects provide details to glue codes for making API calls and validating responses. They are designed to handle multiple method calls for repetitive scenarios.
+
+## Firebolt object
+
+Firebolt objects can be added within JS files located in the `cypress/fixtures/fireboltCalls` folder.
+
+### Sample Firebolt Object Format
+
+Below is the sample format showing how to create a firebolt object.
+
+```
+  JS_OBJECT_NAME = {
+    method: "getMethod name",
+    params: {},
+    validationJsonPath: resolveAtRuntime(["result.{{attribute}}", "result.styles.{{attribute}}"]),
+    setMethod: resolveAtRuntime("manage_closedcaptions.set{{attribute.uppercaseFirstChar}}"),
+    setParams: resolveAtRuntime("value"),
+    setValidationJsonPath: "result",
+    setContent: null,
+    event: "event nameto be validated",
+    eventValidationJsonPath: resolveAtRuntime(["eventResponse.{{attribute}}", "eventResponse.styles.{{attribute}}"]),
+    content: {
+        data: [
+            {
+                type: "fixture",
+                validations: [
+                    {
+                        mode: "staticContentValidation",
+                        type: resolveAtRuntime("value"),
+                        description: resolveAtRuntime("Validating that accessibility.closedCaptionSettings {{attribute}} is {{value}}")
+                    }
+                ]
+            }
+        ]
+    }
+  };
+```
+
+#### Params:
```

Comment:
Can we split apart the the object params and runtimeSpecific functionality? What I mean by that is you can have object with fields like:
```
FIREBOLT_CALL = {
    method: ...,
    params: ...,
    validationJsonPath: ...,
    setMethod: ...,
    setParams: ...,
    setValidationJsonPath: ...,
    setContent: ...,
    event: ...,
    eventValidationJsonPath: ...,
    content: ...
  };
```
Then explain what each field is, what it is used for and an example i.e
| **Param**    | **Definition**  | **Example** |
| ------------ | -------------- | ------------- |
| method        | The method call to make, associated with get test steps | **accessibility.closedCaptionsSettings** |

Also may be good to add a note for acceptable value types i.e string, number, array, object etc... - since some can only be strings and some can be a variety of types.

Additionally we should state that these are all the supported fields but depending on the test scenario and glue code used none of them are technically required.

Reply:
Then after describing the object format explain the dynamic aspect of it. 
Explain how the two fields work:
* attribute
* value

Explain that every field can be a function call `resolveAtRuntime()` and the different ways to use that function call:
* resolveAtRuntime('value')
* resolveAtRuntime('embedded {{value}}')
* resolveAtRuntime(['array', 'with {{value}}s', 'embedded{{attribute}}'])
* resolveAtRuntime('{{attribute. uppercaseFirstChar}}')
* resolveAtRuntime('{{attribute. lowercaseFirstChar}}')

Add a note about how and where FCS will to a deep search for the resolveAtRuntime function (thinking about content but not sure if it was implemented for other fields or not).

Reply:
addressed the comment